src/Polly.SharedSpecs/Timeout/TimeoutAsyncSpecs.cs

```diff
@@ -705,8 +738,381 @@ public void Should_call_ontimeout_but_not_with_task_wrapping_abandoned_action__o
 
             taskPassedToOnTimeout.Should().BeNull();
         }
-        
+
         #endregion
 
+
+        #region onTimeout full argument list - pessimistic
+
+        [Fact]
+        public void Should_call_ontimeout_full_argument_list_with_configured_timeout__pessimistic()
+        {
+            TimeSpan timeoutPassedToConfiguration = TimeSpan.FromMilliseconds(250);
+
+            TimeSpan? timeoutPassedToOnTimeout = null;
+            Func<Context, TimeSpan, Task, Exception, Task> onTimeoutAsync = (ctx, span, task, exception) =>
+            {
+                timeoutPassedToOnTimeout = span;
+                return TaskHelper.EmptyTask;
+            };
+
+            var policy = Policy.TimeoutAsync(timeoutPassedToConfiguration, TimeoutStrategy.Pessimistic, onTimeoutAsync);
+
+            policy.Awaiting(async p => await p.ExecuteAsync(async () =>
+            {
+                await SystemClock.SleepAsync(TimeSpan.FromSeconds(3), CancellationToken.None).ConfigureAwait(false);
+
+            }))
+            .ShouldThrow<TimeoutRejectedException>();
+
+            timeoutPassedToOnTimeout.Should().Be(timeoutPassedToConfiguration);
+        }
+
+        [Fact]
+        public void Should_call_ontimeout_full_argument_list_with_non_null_exception__pessimistic()
+        {
+            TimeSpan timeoutPassedToConfiguration = TimeSpan.FromMilliseconds(250);
+
+            Exception exceptionPassedToOnTimeout = null;
+            Func<Context, TimeSpan, Task, Exception, Task> onTimeoutAsync = (ctx, span, task, exception) =>
+            {
+                exceptionPassedToOnTimeout = exception;
+                return TaskHelper.EmptyTask;
+            };
+
+            var policy = Policy.TimeoutAsync(timeoutPassedToConfiguration, TimeoutStrategy.Pessimistic, onTimeoutAsync);
+
+            policy.Awaiting(async p => await p.ExecuteAsync(async () =>
+            {
+                await SystemClock.SleepAsync(TimeSpan.FromSeconds(3), CancellationToken.None).ConfigureAwait(false);
+
+            }))
+            .ShouldThrow<TimeoutRejectedException>();
+
+            exceptionPassedToOnTimeout.Should().NotBeNull();
+            /*
+                This assertion fails on NET40 as the exception is of type TaskCanceledException
+                Unfortunately, using a #if NET40 switch is not viable as the Polly.Net40Async.Specs 
+                project is using NET45 conditional compilation symbols
+            */
+            // exceptionPassedToOnTimeout.Should().BeOfType(typeof(OperationCanceledException));
```

Comment:
Because we have discontinued .NET4.0 support from v6.x, it is ok now to re-include:

    exceptionPassedToOnTimeout.Should().BeOfType(typeof(OperationCanceledException));

(and remove the comment about .NET4.0 obviously...).

Looks like this applies in all 4 test files. Many thanks!